doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+# Jextract Guide
+
+The jextract tool parses header (.h) files of native libraries, and generates Java code,
+called 'bindings', which  use the Foreign Function and Memory API (FFM API) under the hood,
+that can be used by a client to access the native library.
+
+Interacting with native C code through the FFM API works
+by loading a native library (e.g. a `.so`/`.dll`/`.dylib` file), which is essentially an
+archive of native functions and global variables. The user then has to look up the functions
+they want to call using a `SymbolLookup`, and finally 'link' the functions by using the
+`Linker::downcallHandle` method. Additionally, a client may need to create function pointer for
+Java functions using `Linker::upcallStub`, access global variables through the addresses
+returned by a lookup, and construct `MemoryLayout` instances for the structs they want to access.
+Jextract aims to automate many of these steps, so that a client can instead immediately start
+using a native library they are interested in.
```

Comment:
I suggest using `the native library` (because it's the one they are interested in). Also consider using plural (e.g. the native libraries).